library/core/qsl_base/src/main/java/org/quiltmc/qsl/base/api/util/Maybe.java

```diff
@@ -267,6 +267,15 @@ public static <T> Maybe<T> fromOptional(Optional<T> opt) {
 	 */
 	public abstract T unwrapOrGet(Supplier<? extends T> supplier);
 
+	/**
+	 * Works like {@link Maybe#unwrap} except it throws the provided {@link Throwable} and not the default one.
+	 *
+	 * @param throwableSupplier The {@link Throwable} to throw.
+	 * @return The contained value, if it exists.
+	 * @throws Throwable A customizable throwable.
+	 */
+	public abstract T unwrapOrThrow(Supplier<? extends Throwable> throwableSupplier) throws Throwable;
```

Comment:
```suggestion
	public abstract <E extends Throwable> T unwrapOrThrow(Supplier<? extends E> throwableSupplier) throws E;
```
This prevents checked exceptions from being thrown without javac realizing (or at least, makes it harder to do this accidentally).